### PR TITLE
Burgundy Glider Update

### PR DESCRIPTION
**Pull request by area**
Before we can release a new version of the contracts, ColonyJS needs to know how to talk to them! That's what this PR is... broadly speaking, this comes down to letting it interact correctly with expenditures. The only wrinkle here is that on the contracts, we have two functions that are named `transferExpenditure`. I have added a custom sender that establishes which to used based on whether the address is the owner of the expenditure or not, which assumes that if you are the owner of the expenditure, you wish to use the corresponding function, and will only use the Arbitration permission if you are not the owner.

`transferExpenditureViaOwnership` and `transferExpenditureViaArbitration` are also available for the case that someone implementing a process via ColonyJS wishes to explicitly control which mechanism is used.

This is pretty far outside my wheel house, so fully expecting there to be a lot of improvements. I've also not touched the tests at all...

### CODE REVIEW

File: packages/colony-js-client/src/ColonyClient/senders/AddExpenditure.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* @flow */
+/* eslint-disable import/no-cycle */
+
+import DomainAuth from './DomainAuth';
+
+// XXX This is a good use-case for some kind of async validation step,
+// but since the underlying method functionality is due to change very soon,
+// we're opting to not make big changes to the Sender behaviour, and simply
+// extend the `send` method to perform this async validation.
```
[the remark this answers is not in the document]
Comment (area) [reply packages/colony-js-client/src/ColonyClient/senders/AddExpenditure.js]: Heh, sadly this is copy-and-pasted from an existing file, so I can't claim credit for it ;-)

[the remark this answers is not in the document]
Comment (area) [reply packages/colony-js-client/src/ColonyClient/senders/AddExpenditure.js]: Heh, sadly this is copy-and-pasted from an existing file, so I can't claim credit for it ;-)

**Comment by chmanie** [comment]
I rebased this. @area is this still up-to-date? Do you need to add anything?

**Comment by area** [comment]
Yup, comparing to `develop` in colonyNetwork, I believe this is still up to date.

File: packages/colony-js-client/src/ColonyClient/index.js [hunk 2]
```diff
@@ -1887,6 +2179,57 @@ export default class ColonyClient extends ContractClient {
...
+    ColonyClient,
+    {
+      contract: 'ColonyExpenditure.sol',
+      interface: 'IColony.sol',
+      version: 'burgundy-glider',
+    },
+  >;
+
+  /*
+  Transfer Expenditure as via the arbitration permission
```
Comment (kronosapiens) [inline packages/colony-js-client/src/ColonyClient/index.js]: This comment seems incorrect?

File: packages/colony-js-client/src/ColonyNetworkClient/index.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ import CreateToken from './senders/CreateToken';
 import addRecoveryMethods from '../addRecoveryMethods';
 
 type Address = string;
+type AnyAddress = string;
```
Comment (kronosapiens) [inline packages/colony-js-client/src/ColonyNetworkClient/index.js]: What is this type, above and beyond just a regular `Address`?

Comment (chmanie) [reply packages/colony-js-client/src/ColonyNetworkClient/index.js]: I feel like they used to be different once

File: packages/colony-js-client/src/ColonyClient/index.js (hunk 2 above)
Comment (chmanie) [reply packages/colony-js-client/src/ColonyClient/index.js]: I'll make a PR with some fixes and will include this

**Pull request merged**